test/check-avocado

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2015 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+import unittest
+
+import testvm
+from testlib import *
+
+avocado_tests = ["checklogin-basic.py",
+                 "checklogin-raw.py",
+                ]
+
+
+@unittest.skipIf("atomic" in os.environ.get("TEST_OS", ""), "atomic: missing avocado")
+class TestAvocadoSuite(MachineCase):
+
+    def setUp(self):
+        MachineCase.setUp(self)
+        self.machine_test_dir = "/tmp/avocado_tests"
+        self.m = self.machine
+        self.start_cockpit()
+
+    def testAvocado(self):
+        self.allow_journal_messages("Error receiving data: Connection reset by peer")
+        self.m.execute(command="mkdir -p " + MachineCase.avocado_results_dir)
+        # first upload all of the tests to the target machine
+        self.m.upload([os.path.join(self.m.test_dir, "avocado")],
+                      self.machine_test_dir)
+        # upload sizzle.js to proper dir
+        #self.m.upload([os.path.join(self.m.test_dir, "sizzle.js")],
+        #              self.machine_test_dir)
+
+        cmd_parts = ["avocado run",
+                     "--job-results-dir " + MachineCase.avocado_results_dir,
+                     ' '.join([self.machine_test_dir +
+                               os.sep + x for x in avocado_tests])
+                     ]
+
+        try:
+            self.m.execute(" ".join(cmd_parts))
+        except:
+            self.fail("avocado tests failed")
```

Comment:
This is going to be very irritating. Typically we look at attachments to diagnose the issue, and basic reporting should be inline. Does avocado have something we can show inline here?